mailpoet/assets/js/src/automation/templates/components/template-detail.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  ComponentType,
+  forwardRef,
+  MouseEventHandler,
+  ReactNode,
+  useCallback,
+  useState,
+} from 'react';
+import apiFetch from '@wordpress/api-fetch';
+import { Button, Modal, Snackbar as WpSnackbar } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { chevronLeft, chevronRight, warning } from '@wordpress/icons';
+import { Tag } from '@woocommerce/components';
+import { addQueryArgs } from '@wordpress/url';
+import { TemplatePreview } from './template-preview';
+import { AutomationTemplate, automationTemplateCategories } from '../config';
+import { MailPoet } from '../../../mailpoet';
+
+// snackbar icon is not annotated in the types
+const Snackbar = WpSnackbar as ComponentType<
+  WpSnackbar.Props & {
+    icon: ReactNode;
+  }
+>;
+
+const getCategory = (template: AutomationTemplate): string =>
+  automationTemplateCategories.find(({ slug }) => slug === template.category)
+    ?.name ?? __('Uncategorized', 'mailpoet');
+
+const useCreateFromTemplate = () => {
+  const [state, setState] = useState({
+    data: undefined,
+    loading: false,
+    error: undefined,
+  });
+
+  const create = useCallback(async (slug: string) => {
+    setState((prevState) => ({ ...prevState, loading: true }));
+    try {
+      const data = await apiFetch<{ data: { id: string } }>({
+        path: `/automations/create-from-template`,
+        method: 'POST',
+        data: { slug },
+      });
+      MailPoet.trackEvent('Automations > Template selected', {
+        'Automation slug': slug,
+      });
+      window.location.href = addQueryArgs(MailPoet.urls.automationEditor, {
+        id: data.data.id,
+      });
+    } catch (error) {
+      setState((prevState) => ({ ...prevState, error }));
+    } finally {
+      setState((prevState) => ({ ...prevState, loading: false }));
+    }
+  }, []);
+
+  return [create, state] as const;
+};
+
+type Props = {
+  template: AutomationTemplate;
+  onRequestClose: Modal.Props['onRequestClose'];
+  onPreviousClick?: MouseEventHandler<HTMLButtonElement>;
+  onNextClick?: MouseEventHandler<HTMLButtonElement>;
+};
+
+export const TemplateDetail = forwardRef<HTMLDivElement, Props>(
+  ({ template, onRequestClose, onPreviousClick, onNextClick }, ref) => {
+    const [createAutomationFromTemplate, { loading, error }] =
+      useCreateFromTemplate();
+
+    return (
+      <Modal
+        ref={ref}
+        className="mailpoet-automation-template-detail"
+        title=""
+        onRequestClose={onRequestClose}
+      >
+        <div className="mailpoet-automation-template-detail-content">
+          <div className="mailpoet-automation-template-detail-info">
+            <Tag label={getCategory(template)} />
+            <h1>{template.name}</h1>
+            {template.description}
+          </div>
+          <div className="mailpoet-automation-template-detail-preview">
```

Comment:
I wonder if we should display the scrollbar in the preview only when one is needed?

Reply:
Here is a screenshot:

![Screenshot from 2023-12-01 11-41-14](https://github.com/mailpoet/mailpoet/assets/77215/131e61bd-7068-46fb-90ac-5715aa498086)

I can confirm that switching to `overflow: auto` on `.mailpoet-automation-editor-automation` fixes the issue.